hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -753,6 +787,40 @@ public void updateRpc(MethodDescriptor method, Message param, CallStats stats, T
     updateRpcGeneric(methodName.toString(), stats);
   }
 
+  /** Report table rpc context to metrics system. */
+  private void updateTableMetric(String methodName, TableName tableName,
+    HBaseProtos.RegionSpecifier regionSpecifier, Message param, CallStats stats, Throwable e) {
+    if (tableMetricsEnabled) {
+      if (methodName != null) {
+        String table;
+        if (tableName == null || StringUtils.isEmpty(tableName.getNameAsString())) {
+          // Fallback to get table name from region specifier.
```

Comment:
i think we can simplify this whole method (and above calls) now. I'm not sure we need to handle fallback, since all cases should have a TableName in the controller now. So we don't need to extract the region in updateRpc above, nor do we have to to parse the tablename from the region here.

if we missed a spot, i think we'd consider that a bug and fix it.  maybe you could add end-to-end tests to ensure that each of the request types updates a table metric?

Reply:
@bbeaudreault hi, I have already modified the code according to your requirements. Could you continue the code review for me if you are available? Thanks!